src/providers/index.tsx

```diff
@@ -23,20 +27,22 @@ export const AllProviders = ({
   children,
 }: Props): React.ReactElement => {
   return (
-    <ThemeProvider
-      initialThemeType={initialThemeType}
-      customTheme={{ light, dark }}
-    >
-      <ApolloProvider client={testClient}>
-        <FriendProvider>
-          <AuthProvider initialAuthUser={initialAuthUser}>
-            <ProfileModalProvider>
-              {children}
-            </ProfileModalProvider>
-          </AuthProvider>
-        </FriendProvider>
-      </ApolloProvider>
-    </ThemeProvider>
+    <DeviceProvider>
+      <ThemeProvider
+        initialThemeType={initialThemeType}
+        customTheme={{ light, dark }}
+      >
+        <ApolloProvider client={testClient}>
+          <FriendProvider>
+            <AuthProvider initialAuthUser={initialAuthUser}>
+              <ProfileModalProvider>
+                {children}
+              </ProfileModalProvider>
+            </AuthProvider>
+          </FriendProvider>
+        </ApolloProvider>
```

Comment:
damn...these perplexed contexts are out of the form of beauty. I'll definitely work on manipulating Apollo client state management [[1]](https://medium.com/the-notice-board/life-without-redux-using-apollo-for-local-state-d32b020ff4d3)[[2]](https://engineering.circle.com/https-medium-com-mattdionis-move-over-redux-apollo-client-as-a-state-management-solution-1f9325f96cdd) to contribute getting rid of them.

Reply:
@marsinearth do you mean the nested form of providers? [this might be helpful](https://github.com/facebook/react/issues/14620#issuecomment-491366098)

Reply:
That is a good idea! I hope you can bring up a better code and announce it to us in the meetup.
Also, don't forget to commit `expect` statement you want to accomplish in the previous review.